### PR TITLE
DPL-433 - Fix technical debt for Pacbio Run to ensure we are Revio ready

### LINKED ISSUE #973: DPL-583 Traction: As a user, I want to avoid saving the edited data in a modal pop-up when accidentally closing the modal window by clicking outside the modal display area
**User story**
As a user, I WANT to avoid saving the edited data in a modal pop-up when accidentally closing the modal window by clicking outside the modal display area SO THAT incorrect or draft data is not saved leading to errors in calculation and reporting down the process.

**Who are the primary contacts for this story**
James Watt

**Who is the nominated tester for UAT**
James Watt

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] add a list of acceptance criteria here
- [ ] ...

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by stevieing**
Closes #973 

Changes proposed in this pull request:

*
*
* ...

### CODE REVIEW

File: src/store/traction/pacbio/runCreate/actions.js [hunk 1]
```diff
@@ -58,5 +22,147 @@ export default {
...
+   * @param filter the barcode(s) to find the pools for
+   * @returns { success, errors }. Was the request successful? were there any errors?
+   * TODO: remove getter. Better to use rootState
+   */
+  findPools: async ({ commit, getters }, filter) => {
+    // when users search for nothing, prompt them to enter a barcode
+    if (filter['barcode'].trim() === '') {
+      return {
+        success: false,
+        errors: ['Please provide a plate barcode'],
```
Comment (BenTopping) [inline src/store/traction/pacbio/runCreate/actions.js:39]: 'Please provide a pool barcode'

File: src/store/traction/pacbio/runCreate/run.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
+import Vue from 'vue'
+
+/**
+ * @returns {Object} the default attributes for a run
+ * id new inidcates a new record
+ **/
+const runAttributes = {
```
Comment (BenTopping) [inline src/store/traction/pacbio/runCreate/run.js:7]: defaultRunAttributes may be a better name

File: src/views/pacbio/PacbioRunShow.vue [hunk 3]
```diff
@@ -68,79 +72,44 @@ export default {
...
+    newRecord() {
+      return this.runType.type === RunTypeEnum.New
     },
-    ...mapGetters(['currentRun', 'poolBarcodes']),
-    ...mapState({
-      currentRun: (state) => state.currentRun,
-    }),
+    ...mapGetters(['runType']),
   },
   created() {
```
Comment (BenTopping) [inline src/views/pacbio/PacbioRunShow.vue:82]: I would move this functionality into a method under the methods section. (e.g methods: { fetcher() { ... } }) . Then add the DataFetcher wrapper around the component and pass it in the fetcher method

File: src/views/pacbio/PacbioRunShow.vue [hunk 4]
```diff
@@ -68,79 +72,44 @@ export default {
...
+    alertOnFail({ success, errors }) {
+      if (!success) {
+        this.showAlert(errors, 'danger')
       }
     },
+    // TODO: Still to test through e2e tests.
+    save() {
+      this.saveRun().then(({ success, errors }) => {
+        success
+          ? this.showAlert(`Run successfully created/updated`, 'success', 'run-create-message')
```
Comment (BenTopping) [inline src/views/pacbio/PacbioRunShow.vue:109]: Could change this message based on run type so we don't need created/updated.

File: src/store/traction/pacbio/runCreate/state.js [hunk 5]
```diff
@@ -12,11 +12,31 @@ export default () => {
...
       smrtLinkVersions: {},
     },
+    // Run: The current run being edited or created
+    run: {},
 
+    // Wells: The wells that belong to the run
+    wells: {},
+
+    //Pools: The pools that belong to the wells or the pool selected for a new run
     pools: {},
```
Comment (BenTopping) [inline src/store/traction/pacbio/runCreate/state.js:22]: Just wondering when tubes/libraries/requests/tags are all used. Are they just used to display information in the pool list? If thats the case i am wondering if its worth splitting them all out into separate store objects if they just need to be rejoined anyway. Maybe store all of it nested within the pools? Just a thought, may be a reason we cant do it.

**Review by BenTopping — COMMENTED** [review]
Looking much cleaner, good work 👍  Couple of thoughts / minor corrections. Will do a deeper review when its ready.

File: src/store/traction/pacbio/runCreate/state.js (hunk 5 above)
Comment (stevieing) [reply src/store/traction/pacbio/runCreate/state.js:22]: I would suggest we tackle this as part of the Revio work ...

**Review by BenTopping — APPROVED** [review]

**Comment by BenTopping** [comment]
#973 free win closing this as well.

**Review by BenTopping — APPROVED** [review]

**Pull request merged**